src/utils/OrderedMsgChain.ts

```diff
@@ -16,133 +154,227 @@ const DEFAULT_PROPAGATION_TIMEOUT = 5000
 const DEFAULT_RESEND_TIMEOUT = 5000
 const MAX_GAP_REQUESTS = 10
 
-export type InOrderHandler = (msg: StreamMessage) => void
-export type GapHandler = (from: MessageRef, to: MessageRef, publisherId: string, msgChainId: string) => void
+let ID = 0
 
-export default class OrderedMsgChain extends EventEmitter {
-
-    static MAX_GAP_REQUESTS = MAX_GAP_REQUESTS
-
-    markedExplicitly: WeakSet<StreamMessage>
+class OrderedMsgChain extends MsgChainEmitter {
+    id: number
+    queue = new MsgChainQueue()
+    lastOrderedMsgRef: MessageRef | null = null
+    inProgress: boolean = false
+    gapRequestCount: number = 0
+    maxGapRequests: number = MAX_GAP_REQUESTS
```

Comment:
Could remove the default value as we define it in the constructor

Reply:
👍